Userland/Libraries/LibWeb/Layout/TreeBuilder.cpp

```diff
@@ -572,8 +573,35 @@ void TreeBuilder::generate_missing_parents(NodeWithStyle& root)
             });
         }
 
+        // An anonymous table-wrapper box must be generated around each table-root.
+        if (parent.display().is_table_inside()) {
+            table_roots_to_wrap.append(parent);
+        }
+
         return IterationDecision::Continue;
     });
+
+    for (auto& table_box : table_roots_to_wrap) {
+        auto* nearest_sibling = table_box->next_sibling();
+        auto& parent = *table_box->parent();
+
+        CSS::ComputedValues wrapper_computed_values;
+        // The computed values of properties 'position', 'float', 'margin-*', 'top', 'right', 'bottom', and 'left' on the table element are used on the table wrapper box and not the table box;
+        // all other values of non-inheritable properties are used on the table box and not the table wrapper box.
+        // (Where the table element's values are not used on the table and table wrapper boxes, the initial values are used instead.)
+        static_cast<CSS::MutableComputedValues&>(wrapper_computed_values).set_display(CSS::Display(CSS::Display::Outside::Block, CSS::Display::Inside::FlowRoot));
+        static_cast<CSS::MutableComputedValues&>(wrapper_computed_values).set_position(table_box->computed_values().position());
+        static_cast<CSS::MutableComputedValues&>(wrapper_computed_values).set_inset(table_box->computed_values().inset());
+        static_cast<CSS::MutableComputedValues&>(wrapper_computed_values).set_float(table_box->computed_values().float_());
+        static_cast<CSS::MutableComputedValues&>(wrapper_computed_values).set_margin(table_box->computed_values().margin());
```

Comment:
Instead of doing `static_cast` over and over, please put this in a local:
```cpp
auto& mutable_values = static_cast<CSS::MutableComputedValues&>(wrapper_computed_values);
mutable_values.set...
mutable_values.set...
```